rotkehlchen/tests/unit/test_cost_basis.py

```diff
@@ -406,5 +408,57 @@ def test_reduce_asset_amount_more_that_bought(accountant):
 
     # Also reduce WETH, to make sure it's counted same as ETH
     assert not cost_basis.reduce_asset_amount(A_WETH, FVal(3), 0)
-    acquisitions_num = len(asset_events.acquisitions)
+    acquisitions_num = len(asset_events.acquisitions._acquisitions)
     assert acquisitions_num == 0, 'all buys should be used'
+
+
+def test_accounting_fifo_order(accountant):
```

Comment:
Tests should probably be more exhaustive and I'm going to improve them when the approach will be verified

Reply:
Improved!